bots/ec2_detach_instance_role.py

```diff
@@ -0,0 +1,38 @@
+'''
+## ec2_detach_instance_role
+What it does: Detach an instance role from an EC2 instance.
+Usage: AUTO: ec2_detach_instance_role
+
+Sample GSL: Instance should have roles
+'''
+
+import boto3
+from botocore.exceptions import ClientError
+
+
+def run_action(boto_session, rule, entity, params):
+    text_output = ''
+
+    role_name = entity.get('name')
+
+    iam = boto_session.client('iam')
+    try:
+        response = iam.list_instance_profiles_for_role(RoleName=role_name)['InstanceProfiles']
+        if len(response) == 0:
+            text_output = 'The %s role is not attached to an instance .\nExiting\n' % str(role_name)

```

Comment:
More elegant way:
f'The {role_name} role is not attached to an instance .Exiting'